core/store/store_utils.cpp

```diff
@@ -160,4 +160,38 @@ int64_t bytes_view_input::checksum(size_t offset) const {
   return crc.checksum();
 }
 
+size_t remapped_bytes_view_input::src_to_internal(size_t t) const noexcept {
+  assert(!mapping_.empty());
+  auto it =
+    std::lower_bound(mapping_.begin(), mapping_.end(), t,
+                     [](const auto& l, const auto& r) { return l.first < r; });
+  if (it == mapping_.end()) {
+    --it;
+  } else if (it->first > t) {
+    assert(it != mapping_.begin());
```

Comment:
why is it true?

Reply:
If it is false - remapped bytes input is used in a wrong way. Someone asked for offset that is before first mapping

Reply:
Should not happen. It is a bug  on call site if so